CONTRIBUTING_TO_OPENSHIFT_PLUGINS.md

```diff
@@ -0,0 +1,280 @@
+
+### Submit the new release to the Jenkins organization
+
+Again: your Git ID has push access to the two repositories for this plugin; your Jenkins ID (https://wiki.jenkins-ci.org/display/JENKINS/User+Account+on+Jenkins) is listed in the permission file for the plugin, like https://github.com/jenkins-infra/repository-permissions-updater/blob/master/permissions/plugin-openshift-pipeline.yml.  Given these assumptions:
+
+1. Then run `mvn release:prepare release:perform`
+1. You'll minimally be prompted for the `release version`, `release tag`, and the `new development version`.  Default choices will be provided for each, and the defaults are typically acceptable, so you can just hit the enter key for all three prompts.  As an example, if we are currently at v1.0.36, it will provide 1.0.37 for the new `release version` and `release tag`.  For the `new development version` it will provide 1.0.38-SNAPSHOT, which is again acceptable.  The only time you *might* have to override the default provided is if we currently depend on a SNAPSHOT version of openshift-restclient-java (e.g. `5.3.0-SNAPSHOT`).  This occurs when we add new features to openshift-restclient-java, but the eclipse team has not cut a new, official release (which will typically look like `5.3.0-FINAL`).  If we are in such a mode, you'll get prompted about moving off the SNAPSHOT version (the default provided would be `5.3.0`), but override this (i.e. type in `5.3.0-SNAPSHOT`).	
+1. The `mvn release:prepare release:perform` command will take a few minutes to build the plugin and go through various verifications, followed by a push of the built artifacts up to Jenkins.  This typically works without further involvement but has failed for various reasons in the past.  If so, to retry with the same release version, you'll need to call `git reset` to back of the two commits created as part of publishing the release, as well as use `git tag` to delete both the local and remote version of the corresponding tag.  After deleting the commits and tags, use `git push -f` to update the commits at [https://github.com/jenkinsci/openshift-pipeline-plugin](https://github.com/jenkinsci/openshift-pipeline-plugin). Address whatever issues you have (you might have to solicit help on the Jenkins developer group: https://groups.google.com/forum/#!forum/jenkinsci-dev) and try again.
+
+### Get the new plugin version commits back to the openshift repositories
+
+If the plugin's commits are in sync between openshift and jenkinsci:
+
+* Run `git push https://github.com/openshift/jenkins-plugin.git master` to upload the 2 commits created for cutting the new release to our upstream, development repository, and get the two repositories back in sync.
+
+If not, then you'll have to fetch/cherry-pick this time in the jenkinsci to openshift direction.
```

Comment:
what's the consequence of not putting the release commits into the openshift repository?  Since we never cut releases from that repository, does it even matter?

Reply:
If we want to maintain the `git rebase` option for syncing the 2 repositories - I'll clarify... if we have punted on that approach, yes, it is not critical, though I find it still a useful approximation / sanity check.

Reply:
It is also helpful when building patches for users, to lend some sanity to which versions the patches are in between ... again, I'll attempt to clarify in the upcoming push.